WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsAndTopPerformersViewController.swift

```diff
@@ -117,59 +117,54 @@ private extension StoreStatsAndTopPerformersViewController {
             }
         }
 
-        // Loads site for the latest site timezone.
-        let loadSiteAction = AccountAction.loadSite(siteID: siteID) { [weak self] site in
-            guard let site = site, let self = self else {
-                return
-            }
-            let timezone = site.timezone
-            let siteTimezone = TimeZone(identifier: timezone) ?? TimeZone(secondsFromGMT: 0) ?? TimeZone.current
-            self.periodVCs.forEach { (vc) in
-                vc.siteTimezone = siteTimezone
-
-                let currentDate = Date()
-                vc.currentDate = currentDate
-                let latestDateToInclude = vc.timeRange.latestDate(currentDate: currentDate, siteTimezone: siteTimezone)
-
-                group.enter()
-                self.syncStats(for: siteID,
-                               siteTimezone: siteTimezone,
-                               timeRange: vc.timeRange,
-                               latestDateToInclude: latestDateToInclude) { [weak self] error in
-                    if let error = error {
-                        DDLogError("⛔️ Error synchronizing order stats: \(error)")
-                        syncError = error
-                    } else {
-                        self?.trackStatsLoaded(for: vc.granularity)
-                    }
-                    group.leave()
+        // When the site time zone can be correctly fetched, and also supports the Stats v4 API, consider using the site time zone
+        // for stats UI (#1375).
+        let timezoneForStatsDates = TimeZone.current
```

Comment:
Hey Jaclyn 👋 I'm just now catching up. So if I understand your solution: this is taking the timezone from the device, calculate the time range for the API request, and using that response for the v4 stats. Per the conversation here p1571387404105600-slack-C6H8C3G23, it sounds like Android is using the site time zone and not the device time zone. 

So if I follow all the conversations correctly, that means this PR can be closed without merging, right?

Reply:
> Android is using the site time zone and not the device time zone.

Sorry the API setup is quite complicated. **Using the site time zone = not specifying any time zone = device time zone from the app's perspective** (not setting time zone manually on a device = dates are in the device time zone). This means that in the API requests, we don't specify any time zone in the dates we send up, and we use the dates in the site time zone (**not in GMT**) from the API response to display in UI.

In the stats v4 API response, there are two types of dates - one GMT, the other one in the site time zone:

```
{
                "interval": "2019-10-12",
                "date_start": "2019-10-12 00:00:00",
                "date_start_gmt": "2019-10-11 23:00:00",
                "date_end": "2019-10-12 23:59:59",
                "date_end_gmt": "2019-10-12 22:59:59",
                ...
}
```

Before this PR, we used the GMT dates (`date_start_gmt`/`date_end_gmt`) plus the site time zone GMT offset to render the stats intervals in the site time zone. This PR now uses the dates in the site time zone from the API response (`date_start`/`date_end`) and render these dates without any time zone offset (thus using the device time zone here). Please let me know if removing the whole time zone code here might make everything more clear (was trying to fix this issue with less code)!

> So if I follow all the conversations correctly, that means this PR can be closed without merging, right?

Hopefully the above explanation makes sense, and that means this PR is needed to fix the current issue where the UTC-based sites don't have a correct site time zone in the app (there are a few issues with the site time zone, one in the app and the other in the API) by not using the site time zone anymore.

Reply:
So you're saying: (True or False?)

1. The `site.timezone` always reports as an empty string if the user has never specified the timezone in Settings > General > Timezone. (And you can tell that the user has never set it, because the drop-down menu in the web site shows UTC+0.)
2. When the site doesn't specify a timezone, you previously compensated for that by getting the device's current timezone.
3. You were using `date_start_gmt` and `date_end_gmt`, which is also was wrong, because without a proper timezone, GMT is just making things up.
4. We need **something, anything** that gives us some kind of anchor for what their true time zone is, so you've skipped the check for the `site.timezone` and went straight to asking device. 

Did I get it right this time? (I'm really sorry about this, I have to have date / time explained like I'm 5 years old, because I'm seriously that slow with dates & times.)
If I got all that, is there any harm in keeping the initial check for the site.timezone (in case it's ever, you know, not empty)?

Reply:
Thanks for following up with the questions! I hope the answers help explain the problems and the changes in this PR, and we can do a video call if it's easier :)

> 1. The site.timezone always reports as an empty string if the user has never specified the timezone in Settings > General > Timezone. (And you can tell that the user has never set it, because the drop-down menu in the web site shows UTC+0.)

A site's `timezone` field from the WP.com `/me/sites` API (which we use to retrieve various site settings) is empty if the site time zone is UTC based (UTC+offset). For sites in UTC-based time zone, we'd have to get the time zone from `gmt_offset` field in the API. Otherwise, for time zones specified by a region (e.g `America/Log Angeles`) the site time zone is correct in the API's `timezone` field. I don't know which time zone is the default, it'd be either region or UTC based.

Currently, we only fetch the `timezone` field in the API, and that's why it works for sites whose time zone is region-based but not the sites with UTC-based time zone (like the site in #1357). I only learned about this while debugging #1357 !

<img width="621" alt="Screen Shot 2019-10-23 at 9 44 08 AM" src="https://user-images.githubusercontent.com/1945542/67349519-b6035e00-f57a-11e9-87b1-6772b7c12370.png">

> 2. When the site doesn't specify a timezone, you previously compensated for that by getting the device's current timezone.

I'd interpret "when the site doesn't specify a timezone" as "when the site's time zone is UTC based": in this case, it can't read `site.timezone` on iOS, so it defaults to GMT+0 (code [here](https://github.com/woocommerce/woocommerce-ios/blob/develop/WooCommerce/Classes/ViewRelated/Dashboard/Stats%20v4/StoreStatsAndTopPerformersViewController.swift#L126)). When I worked on stats v4, I didn't know that `site.timezone` could be empty and how the UTC-based sites work as explained in your question 1.

> 3. You were using date_start_gmt and date_end_gmt, which is also was wrong, because without a proper timezone, GMT is just making things up.

Using GMT dates is correct **if we have the correct site time zone** for the GMT offset. Like when we talk about different time zones, we usually say GMT with an offset so it's easy to calculate the time zone difference. I generally prefer GMT dates, because it's at an objective time. When we use the dates in the site time zone, we don't know when each date actually starts.

The issues with the GMT dates on iOS:
1) First, we need to get the correct time zone for UTC-based sites. To achieve this, we'd have to also get the `gmt_offset` field from the site settings API from your question 1. I created an issue here: https://github.com/woocommerce/woocommerce-ios/issues/1375
2) After we have the correct site time zone, there's still an existing issue with stats v4 API when the GMT offset is not an integer: 
(a core issue) https://github.com/woocommerce/woocommerce/pull/24863

Since the time zone API fix wouldn't come immediately, I think it's better to fix #1357 by just using the dates in the site time zone from the API (by sending and displaying all the dates from the API with device time zone).

> 4. We need something, anything that gives us some kind of anchor for what their true time zone is, so you've skipped the check for the site.timezone and went straight to asking device.

In the stats v4 UI case, we don't need to know what the true site time zone is because all **the stats are displayed relative to the site time zone** as on WC Admin web version. This wouldn't be the case if we ever want to display the stats in the device time zone (because the device time zone could be different from the site time zone).

Regarding the reason for using the device time zone in this PR: every `Date` on iOS has an absolute timestamp (like when we call [`timeIntervalSince1970`](https://developer.apple.com/documentation/foundation/nsdate/1407504-timeintervalsince1970)), which is a different local time depending on the time zone. When we are using the dates in the site time zone from the API, the API gives us a date like `"date_start": "2019-10-12 00:00:00"` and we want to display this start date on the UI. We don't know what the site time zone is and it doesn't matter, because the UI is all relative to the site time zone. But to create a `Date` on iOS from a date string like `2019-10-12 00:00:00`, a time zone is used (and default to the device time zone) to create an absolute date. In the UI, we are seeing the date as `2019-10-12 00:00:00` from a `Date` object in the device time zone, while the date should be interpreted as in the site time zone by the user. This is how WC Admin works on the web if your site is in a different time zone from the device/browser time zone.

> is there any harm in keeping the initial check for the site.timezone (in case it's ever, you know, not empty)?

From the answer to your question 3, we can't use the GMT dates + site time zone's GMT offset to calculate the dates for stats v4 anymore. From the answer to your question 4, we don't need to know the site time zone if we're always displaying the stats relative to the site time zone. Therefore, a check for the site time zone isn't necessary anymore and that's why I removed it.